lib/puppet/type/ios_ntp_config.rb

```diff
@@ -0,0 +1,30 @@
+require 'puppet/resource_api'
+Puppet::ResourceApi.register_type(
+  name: 'ios_ntp_config',
+  docs: 'Specify NTP config',
+  features: ['canonicalize', 'simple_get_filter'] + (Puppet::Util::NetworkDevice.current.nil? ? [] : ['remote_resource']),
```

Comment:
The `Puppet::Util::NetworkDevice.current.nil? ` check is redundant in this case, as this module will always be running in `device` mode. So `features: ['canonicalize', 'simple_get_filter', 'remote_resource']` is sufficient.